ziti/enroll/enroll.go

```diff
@@ -269,31 +269,39 @@ func enrollOTT(token *config.EnrollmentClaims, cfg *config.Config, caPool *x509.
 		return err
 	}
 
-	respBody, err := ioutil.ReadAll(resp.Body)
-	if resp.StatusCode != http.StatusOK || err != nil {
-		var edgeError edge.EdgeControllerApiError
-		jsonerr := json.Unmarshal(respBody, &edgeError)
+	body, err := ioutil.ReadAll(resp.Body)
 
-		if jsonerr != nil {
-			return errors.Errorf("enroll error: %s. raw response: %s. ", resp.Status, respBody)
-		}
+	if err != nil {
+		return errors.Errorf("enroll error: %s: could not read body: %s", resp.Status, body)
+	}
 
-		//successfully parsed the json - make sure the code and cause message match and give a helpful message
-		if edgeError.Error.Cause.Message == "identity not found" && edgeError.Error.Code == "INVALID_FIELD" {
-			return errors.Errorf("enroll error: %s. response: the provided token was invalid. either the token is not correct or it has already been used", resp.Status)
-		} else {
-			cause := edgeError.Error.Cause.Message
+	if resp.StatusCode == http.StatusOK {
+		cfg.ID.Cert = "pem:" + string(body)
+		return nil
+	}
 
-			if cause == "" {
-				//if Cause.Message is empty - try using CauseMessage instead
-				cause = edgeError.Error.CauseMessage
-			}
+	jsonErr, err := gabs.ParseJSON(body)
 
-			return errors.Errorf("enroll error: %s. response: %s. ", resp.Status, cause)
+	if err != nil {
+		return errors.Errorf("enroll error: %s: could not parse body: %s", resp.Status, body)
+	}
+
+	if jsonErr.Exists("error", "message") {
+		message := jsonErr.Search("error", "message").Data().(string)
+		code := jsonErr.Search("error", "code").Data().(string)
+
+		//todo: remove causeMessage support when removed from API
+		cause := "unspecified"
```

Comment:
Added cause handling back in here. I plan on removing `causeMessage` in netfoundry/ziti-edge#78